File: fixtures/scale/ruby/large/lib/check_fnvhash.rb
require_relative 'fnvhash_digest'
require_relative 'fnvhash_digest_pair'
require_relative 'corpus_hash'

def check_fnvhash
  corpus_hash.each do |text|
    first = fnvhash_digest(text)
    second = fnvhash_digest(text)
    return false if first != second || first.length != 8
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_fnvhash.py
from cipherlab.ciphers.fnvhash import fnvhash_digest
from cipherlab.specs.fnvhash_spec import fnvhash_spec
from cipherlab.corpus.corpus_hash import corpus_hash


def check_fnvhash():
    spec = fnvhash_spec()
    for text in corpus_hash():
        first = fnvhash_digest(text)
        second = fnvhash_digest(text)
        if first != second or len(first) != 8:
            return False
    return spec["category"] == "hash"
